providers/edge3/src/airflow/providers/edge3/plugins/www/src/components/OperationsCell.tsx

```diff
@@ -0,0 +1,138 @@
+
+interface MaintenanceFormProps {
+  onSubmit: (comment: string) => void;
+  onCancel: () => void;
+}
+
+const MaintenanceForm = ({ onCancel, onSubmit }: MaintenanceFormProps) => {
+  const [comment, setComment] = useState("");
+
+  const handleSubmit = () => {
+    if (comment.trim()) {
+      onSubmit(comment.trim());
+    }
+  };
+
+  return (
+    <VStack gap={2} align="stretch">
+      <Textarea
+        placeholder="Enter maintenance comment (required)"
+        value={comment}
+        onChange={(e) => setComment(e.target.value)}
+        required
+        maxLength={1024}
+        size="sm"
+      />
+      <HStack gap={2}>
+        <IconButton
+          size="sm"
+          colorScheme="green"
+          onClick={handleSubmit}
+          disabled={!comment.trim()}
+          aria-label="Confirm Maintenance"
+        >
+          <FcCheckmark />
+        </IconButton>
+        <IconButton size="sm" colorScheme="red" variant="outline" onClick={onCancel} aria-label="Cancel">
+          <ImCross />
+        </IconButton>
+      </HStack>
+    </VStack>
+  );
+};
+
+interface OperationsCellProps {
+  worker: Worker;
+  activeMaintenanceForm: string | null;
+  onSetActiveMaintenanceForm: (workerName: string | null) => void;
+  onRequestMaintenance: (workerName: string, comment: string) => void;
+  onExitMaintenance: (workerName: string) => void;
+}
+
+export const OperationsCell = ({
+  activeMaintenanceForm,
+  onExitMaintenance,
+  onRequestMaintenance,
+  onSetActiveMaintenanceForm,
+  worker,
+}: OperationsCellProps) => {
+  const workerName = worker.worker_name;
+  const state = worker.state;
+
+  let cellContent = null;
+
+  if (state === "idle" || state === "running") {
+    if (activeMaintenanceForm === workerName) {
+      cellContent = (
+        <MaintenanceForm
+          onSubmit={(comment) => onRequestMaintenance(workerName, comment)}
+          onCancel={() => onSetActiveMaintenanceForm(null)}
+        />
+      );
+    } else {
+      cellContent = (
+        <Flex justifyContent="end">
+          <IconButton
+            size="sm"
+            variant="ghost"
+            onClick={() => onSetActiveMaintenanceForm(workerName)}
+            aria-label="Enter Maintenance"
+          >
+            <HiOutlineWrenchScrewdriver />
+          </IconButton>
+        </Flex>
+      );
+    }
+  } else if (
+    state === "maintenance pending" ||
+    state === "maintenance mode" ||
+    state === "maintenance request" ||
+    state === "maintenance exit" ||
+    state === "offline maintenance"
+  ) {
+    cellContent = (
+      <VStack gap={2} align="stretch">
+        <Box fontSize="sm" whiteSpace="pre-wrap">
+          {worker.maintenance_comments || "No comment"}
+        </Box>
+        <Flex justifyContent="end">
+          <IconButton
+            size="sm"
+            variant="ghost"
+            onClick={() => onExitMaintenance(workerName)}
+            aria-label="Exit Maintenance"
```

Comment:
Replication of commit https://github.com/apache/airflow/commit/fdd0c2b0f7eeeda79a8e00cec50a7417ef0f9406 (3)
```suggestion
            aria-label="Exit Maintenance"
            title="Exit Maintenance"
```